llvm/include/llvm/ProfileData/MemProfSummaryBuilder.h

```diff
@@ -0,0 +1,51 @@
+//===- MemProfSummaryBuilder.h - MemProf summary building -------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file contains MemProf summary builder.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_PROFILEDATA_MEMPROFSUMMARYBUILDER_H
+#define LLVM_PROFILEDATA_MEMPROFSUMMARYBUILDER_H
+
+#include "llvm/ProfileData/MemProf.h"
+#include "llvm/ProfileData/MemProfSummary.h"
+
+namespace llvm {
+namespace memprof {
+
+class MemProfSummaryBuilder {
+private:
+  // The set of full context IDs that we've recorded so far. This is needed to
+  // dedup the MIBs, which are duplicated between functions containing inline
+  // instances of the same allocations.
+  DenseSet<uint64_t> Contexts;
+
+  void addRecord(uint64_t, const PortableMemInfoBlock &);
+
+protected:
```

Comment:
Why are these protected? I don't see a usage where this is needed. Did I miss something?